iceoryx_posh/test/moduletests/test_roudi_mempool_introspection.cpp

```diff
@@ -149,45 +165,45 @@ class MemPoolIntrospection_test : public Test
 
     MePooMemoryManager_MOCK m_rouDiInternalMemoryManager_mock;
     SegmentManagerMock m_segmentManager_mock;
-    SenderPort_MOCK m_senderPortImpl_mock;
+    MockPublisherPortUser m_publisherPortImpl_mock;
+
+    iox::mepoo::MemoryManager m_memoryManager;
+    iox::capro::ServiceDescription m_serviceDescription;
+    iox::popo::PublisherPortData m_publisherPortData{m_serviceDescription, "Foo", &m_memoryManager};
 };
 
 TEST_F(MemPoolIntrospection_test, CTOR)
 {
-    auto mock = m_senderPortImpl_mock.details;
-
     {
-        MemPoolIntrospection m_introspection(
-            m_rouDiInternalMemoryManager_mock, m_segmentManager_mock, std::move(m_senderPortImpl_mock));
-    }
+        MemPoolIntrospectionAccess introspectionAccess(
+            m_rouDiInternalMemoryManager_mock, m_segmentManager_mock, std::move(m_publisherPortImpl_mock));
 
-    EXPECT_THAT(mock->activate, Eq(1));
-    EXPECT_THAT(mock->deactivate, Eq(1));
+        // @todo how to check for offer()? EXPECT_CALL must be set before the object is created?!
```

Comment:
@mossmaurice EXPECT_CALL needs the object where the call happens, therefore you cannot use is for stuff that happens in the c'tor. There was a reason why we did not use the real objects but just mocks. Whit the changes in this PR, this test becomes more of an integration test instead of a module test. I would suggest to revert the changes.

Reply:
Done and works as suggested! :)